framework/include/utils/FunctionMaterialPropertyDescriptor.h

```diff
@@ -108,4 +108,7 @@ class FunctionMaterialPropertyDescriptor : public DerivativeMaterialPropertyName
 
   /// cache the final property name of the derivative
   std::string _property_name;
+
+  /// is this a property that was directly specified by the user? Then it _must_ exist.
+  bool _required;
```

Comment:
```suggestion
  const bool _required;
```
ya?

Reply:
nope

Reply:
To elaborate: It won't compile with `const` because the assignment operator will be implicitly deleted.